r4/src/main/java/org/opencds/cqf/r4/providers/PlanDefinitionApplyProvider.java

```diff
@@ -132,9 +122,40 @@ private void resolveDefinition(Session session, PlanDefinition.PlanDefinitionAct
             logger.debug("Resolving definition " + action.getDefinitionCanonicalType().getValue());
             String definition = action.getDefinitionCanonicalType().getValue();
             if (definition.startsWith(session.getPlanDefinition().fhirType())) {
-                logger.error("Currently cannot resolve nested PlanDefinitions");
-                throw new NotImplementedException(
-                        "Plan Definition refers to sub Plan Definition, this is not yet supported");
+                IdType id = new IdType(definition);
+                CarePlan plan = null;
+                try {
+                    plan = applyPlanDefinition(id,
+                            session.getPatientId(),
+                            session.getEncounterId(),
+                            session.getPractitionerId(),
+                            session.getOrganizationId(),
+                            session.getUserType(),
+                            session.getUserLanguage(),
+                            session.getUserTaskContext(),
+                            session.getSetting(),
+                            session.getSettingContext());
+                } catch (IOException e) {
+                    e.printStackTrace();
+                } catch (JAXBException e) {
+                    e.printStackTrace();
+                }
+                if (plan != null) {
+                    // iterate over nested plans and get contained activities
+                    for(Resource r: plan.getContained())
+                    {
+                        session.getCarePlanBuilder().buildContained(r).buildActivity(
+                                new CarePlanActivityBuilder().buildReference(new Reference("#" + r.getId())).build());
+                    }
+                    for(CanonicalType c: plan.getInstantiatesCanonical())
+                    {
+                        session.getCarePlanBuilder().buildInstantiatesCanonical(c.getValueAsString());
+                    }
+                }
+                else
+                {
+                    logger.error("nested plan failed");
+                }
```

Comment:
Would it make sense to move the contents of the `if` into the `try` above?  That way we can keep all the happy path logic together and remove this seemingly unnecessary `if` statement.  The logger statement in the `else` could be similarly be moved up into the `catch` block.

(Hard to see in GH code review, but its just shuffling things around)

```suggestion
                try {
                    plan = applyPlanDefinition(id,
                            session.getPatientId(),
                            session.getEncounterId(),
                            session.getPractitionerId(),
                            session.getOrganizationId(),
                            session.getUserType(),
                            session.getUserLanguage(),
                            session.getUserTaskContext(),
                            session.getSetting(),
                            session.getSettingContext());

                    // iterate over nested plans and get contained activities
                    for(Resource r: plan.getContained())
                    {
                        session.getCarePlanBuilder().buildContained(r).buildActivity(
                            new CarePlanActivityBuilder().buildReference(new Reference("#" + r.getId())).build());
                    }
                    for(CanonicalType c: plan.getInstantiatesCanonical())
                    {
                        session.getCarePlanBuilder().buildInstantiatesCanonical(c.getValueAsString());
                    }
                } catch (IOException | JAXBException e) {
                    e.printStackTrace();
                    logger.error("nested plan failed");
                }
```